### PR TITLE
http_test.go : Added tests 

### PR DESCRIPTION
**Pull request by sladyn98**
Added tests for http_test.go.
Increased coverage to 12.3%

### CODE REVIEW

**Comment by glaslos** [comment]
Please resolve the conflict

**Comment by sladyn98** [comment]
Do you want me to rename this branch. Or I could even rebase onto your branch which is a bit difficult.

**Comment by glaslos** [comment]
rebase or merge

**Comment by sladyn98** [comment]
Cool I will try and  rebase this branch asap.

**Comment by glaslos** [comment]
Please fix the tests

**Comment by sladyn98** [comment]
Travis fails due to some problems in the previous test in the http test handler.Because I tested my branch on my local machine and tests were passing.

File: http_test.go [hunk 1]
```diff
@@ -33,3 +36,22 @@ func TestHTTPHandler(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+func TestFormatRequest(t *testing.T) {
```
Comment (glaslos) [inline http_test.go]: It doesn't really make sense to test a function by reimplementing it. How about having a fixed output and then you test if the value returned by the function matches?

**Comment by glaslos** [comment]
Why are you merging into the http_test branch? This is a branch that might not be used. Please create a PR for master.

File: http_test.go (hunk 1 above)
Comment (sladyn98) [reply http_test.go]: Okay thanks for the feedback. I will reimplement the tests :+1:

**Comment by sladyn98** [comment]
I tried rebasing onto your branch because I thought you wanted to merge this branch into master.Anyway I will just do a rebase and resolve conflicts :+1:

**Comment by glaslos** [comment]
Please resolve the conflict. I really don't like you doing force pushes. I deletes history and I can't see what changes you made.

**Pull request closed without merging**